apps/src/netsim/NetSimTable.js

```diff
@@ -68,51 +68,75 @@ var NetSimTable = module.exports = function (storageTable) {
   this.pollingInterval_ = DEFAULT_POLLING_DELAY_MS;
 };
 
+/**
+ * @param {!NodeStyleCallback} callback
+ */
 NetSimTable.prototype.readAll = function (callback) {
-  this.remoteTable_.readAll(function (data) {
-    callback(data);
-    if (data !== null) {
+  this.remoteTable_.readAll(function (err, data) {
+    callback(err, data);
+    if (err === null) {
       this.fullCacheUpdate_(data);
     }
   }.bind(this));
 };
 
+/**
+ * @param {!number} id
+ * @param {!NodeStyleCallback} callback
+ */
 NetSimTable.prototype.read = function (id, callback) {
-  this.remoteTable_.read(id, function (data) {
-    callback(data);
-    if (data !== undefined) {
+  this.remoteTable_.read(id, function (err, data) {
+    callback(err, data);
+    if (err === null) {
       this.updateCacheRow_(id, data);
```

Comment:
Specifically, this is the shim between the storage API and the rest of the application which does local caching and provides change notifications.  The work we're doing after the callback is the change notification out to the rest of the system; if it wasn't being done immediately here, those notifications would come in later when the change came down from remote anyway.  I wanted to simulate that behavior as closely as possible by letting the initiating operation complete before kicking off the change notifications to other parts of the system.